cli/lib/cypress.js

```diff
@@ -68,6 +68,19 @@ const cypressModuleApi = {
       return cli.parseRunCommand(args)
     },
   },
+
+  /**
+   * This function should not be used in pure JavaScript
+   * By the type of its argument it allows autocompletion and
+   * type checking of the configuration given by users.
```

Comment:
I don't think I get why the first line of this comment is necessary. I did a quick test and the autosuggestion still works for me in vscode inside of a "pure Javascript" file:
![image](https://user-images.githubusercontent.com/1151760/135508056-8ed19c83-0d7e-43cb-967b-314e7fb4bcdd.png)

I'm interpreting the comment as "don't use this outside of TypeScript".

So should `defineConfig` not be used in `.js` for some reason? Am I misunderstanding what "pure JavaScript" means?

Since this comment is visible to `.js` users I think rewording it to match `cypress-npm-api.d.ts` would make sense:

```suggestion
     * Type helper to make writing `cypress.config.js` easier
```

Reply:
@cowboy My english/code culture/knowhow is lacking here to defend the choice I have made.
Could you advise on what the comment should say please?

Reply:
Since the behavior is the same for `.js` and `.ts` and this is the only clean way (AFAIK) to supply types/autosuggestion for an export, I think we should at least remove the first line of this comment, and probably sync it with `cypress-npm-api.d.ts`'s type export, whatever the comment ends up being.

Reply:
This is the way we're going to be talking about it in the new docs, so try to get your language close to this: https://github.com/cypress-io/cypress-documentation/pull/4131/files#diff-2ddd6ad3e6f0e14719d47ad1df70d747c592bd20195c7971b23f0669b2547a95R43

Eg.

The `defineConfig` helper function is exported by Cypress, and it provides automatic code completion for configuration in many popular code editors. While it's not strictly necessary for Cypress to parse your configuration, we recommend wrapping your config object with `defineConfig()` like this:

Reply:
```suggestion
   * provides automatic code completion for configuration in many popular code editors. 
   * While it's not strictly necessary for Cypress to parse your configuration, we
   * recommend wrapping your config object with `defineConfig()`
   * @example
   * module.exports = defineConfig({
   *   viewportWith: 400
   * })
```